### ISSUE TITLE
inotify example code doesn't actually work

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4399]
|<img alt="benjamin's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| benjamin reported|
|-|-|
|Trac ID|trac#4399|
|Type|defect|
|Created|2010-04-15 02:33:03Z|

Running the code in the inotify.py docstring gives

```
Traceback (most recent call last):
  File "x.py", line 10, in <module>
    notifier.watch("twisted", callbacks=[notify], recursive=True)
  File "/home/benjamin/dev/repos/twisted/twisted/internet/inotify.py", line 346, in watch
    for child in path.walk():
AttributeError: 'str' object has no attribute 'walk'
```

Looking past this issue, it seems the callback accepts too many args.

Attachments:

* [inotify.py-4399.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c6b/c6b2f07c09af378448bc470ef19a8e9523f5ce95/330c5a264a5083617b44eb6963830b2d18965e55.patch) (911 bytes) - added by aoverbury on 2010-04-26 08:02:19Z - Patch to correct the example code in inotify.py so that it actually runs

<details><summary>Searchable metadata</summary>

```
trac-id__4399 4399
type__defect defect
reporter__benjamin benjamin
priority__high high
milestone__Twisted_10_1 Twisted-10.1
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation__easy documentation, easy
time__1271298783000000 1271298783000000
changetime__1272289737000000 1272289737000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

fixing description markup

**Comment by twisted-trac** [comment]
|<img alt="aoverbury's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|aoverbury removed owner|
|-|-|

The problem with the example code is that it was passing in the path to be watched as a string, when the code was expecting a FilePath object from twisted.python.filepath.  It should also probably be noted in the example code that this just sets a watch on the path -- you need to run notifier.doRead() in a loop in order to actually get any notification events out of it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Thanks.  This looks good, and your point about needing a reactor to be running is a good one.  I'm going to add a note about that and then apply this.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [28866]) Apply inotify module docstring example fix patch

Author: aoverbury
Reviewer: exarkun
Fixes: [#4399](https://github.com/twisted/twisted/issues/4399)

Fix the inotify example in the `twisted.internet.inotify` module docstring
to use the `watch` and callback APIs properly (they accept `FilePath`
instances, not `str` instances).  Also drop the unused `extraArg` parameter.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
